tests/kernel/mem_protect/mem_protect/src/mem_partition.c

```diff
@@ -7,9 +7,9 @@
 #include <kernel.h>
 #include "mem_protect.h"
 
-static K_APP_DMEM(ztest_mem_partition) int var = 1356;
-static K_APP_BMEM(ztest_mem_partition) int zeroed_var = 20420;
-static K_APP_BMEM(ztest_mem_partition) int bss_var;
+static volatile K_APP_DMEM(ztest_mem_partition) int var = 1356;
```

Comment:
Can you add some comments to indicate why the 'volatile' was added here?
That's to avoid users copy the example and leave the volatile while in their application it might not be required, it is only for this test.

Reply:
added.